src/Runner.Common/HttpClientHandlerFactory.cs

```diff
@@ -0,0 +1,23 @@
+using System.Net.Http;
+using GitHub.Runner.Sdk;
+
+namespace GitHub.Runner.Common
+{
+    [ServiceLocator(Default = typeof(HttpClientHandlerFactory))]
+    public interface IHttpClientHandlerFactory : IRunnerService
+    {
+        HttpClientHandler CreateClientHandler(RunnerWebProxy webProxy);
+    }
+
+    public class HttpClientHandlerFactory : IHttpClientHandlerFactory
```

Comment:
`public HttpClientHandlerFactory : RunnerService, IHttpClientHandlerFactory`

check other services for example.